tonic-build/src/lib.rs

```diff
@@ -304,32 +227,14 @@ fn generate_doc_comment(comment: &str) -> TokenStream {
 }
 
 // Generate a larger doc comment composed of many lines of doc comments
-fn generate_doc_comments<T: AsRef<str>>(comments: &[T]) -> TokenStream {
+fn generate_doc_comments<'a, T: AsRef<str> + 'a, I: Iterator<Item = &'a T>>(
+    comments: I,
+) -> TokenStream {
     let mut stream = TokenStream::new();
 
-    for comment in comments {
-        stream.extend(generate_doc_comment(comment.as_ref()));
+    for ref comment in comments {
```

Comment:
```suggestion
    for comment in &comments {
```

Reply:
Actually `comments` is an iterator. So, your suggestion is not appliable straightforward. I changed the type of comments to IntoIterator.